File: app/src/test/java/com/lasercyber/lws/ui/bean/http/CameraVideoOverlayEditorTest.java
package com.lasercyber.lws.ui.bean.http;

import com.google.gson.JsonObject;

import org.junit.Assert;
import org.junit.Test;

public class CameraVideoOverlayEditorTest {

    @Test
    public void applyNameOverlay_enable1_setsPositionAndName() {
        JsonObject root = overlayRoot();
        JsonObject updated = CameraVideoOverlayEditor.applyNameOverlay(root, 1, 20, 30, "Laser-01");
        Assert.assertNotNull(updated);
        JsonObject nameOverlay = updated.getAsJsonObject("VideoOverlay").getAsJsonObject("NameOverlay");
        Assert.assertEquals(1, nameOverlay.get("enable").getAsInt());
        Assert.assertEquals(20, nameOverlay.get("x").getAsInt());
        Assert.assertEquals(80, nameOverlay.get("y").getAsInt());
        Assert.assertEquals("Laser-01", nameOverlay.get("name").getAsString());
    }

    @Test
    public void applyNameOverlay_enable0_hidesName() {
        JsonObject root = overlayRoot();
        JsonObject updated = CameraVideoOverlayEditor.applyNameOverlay(root, 0, 10, 10, "HGDevice");
        Assert.assertNotNull(updated);
        Assert.assertEquals(0,
                updated.getAsJsonObject("VideoOverlay")
                        .getAsJsonObject("NameOverlay")
                        .get("enable")
                        .getAsInt());
    }

    @Test
    public void parseOverlayConfig_rejectsErrCodeWrapper() {
        JsonObject error = new JsonObject();
        error.addProperty("errCode", 400);
        Assert.assertNull(CameraVideoOverlayEditor.parseOverlayConfig(error));
    }

    private static JsonObject overlayRoot() {
        JsonObject root = new JsonObject();
        JsonObject videoOverlay = new JsonObject();
        videoOverlay.add("NameOverlay", new JsonObject());
        root.add("VideoOverlay", videoOverlay);
        return root;
    }
}

File: app/lws_hmi/test/camera_video_overlay_editor_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:lws_hmi/features/ip_camera/application/camera_video_overlay_editor.dart';

void main() {
  Map<String, Object?> overlayRoot() => <String, Object?>{
        'VideoOverlay': <String, Object?>{
          'NameOverlay': <String, Object?>{},
        },
      };

  group('CameraVideoOverlayEditor', () {
    test('applyNameOverlay enable=1 sets position and name', () {
      final updated = CameraVideoOverlayEditor.applyNameOverlay(
        overlayRoot(),
        enable: 1,
        positionX: 20,
        positionY: 30,
        name: 'Laser-01',
      );
      expect(updated, isNotNull);
      final nameOverlay = (updated!['VideoOverlay'] as Map)['NameOverlay'] as Map;
      expect(nameOverlay['enable'], 1);
      expect(nameOverlay['x'], 20);
      expect(nameOverlay['y'], 80);
      expect(nameOverlay['name'], 'Laser-01');
    });

    test('applyNameOverlay enable=0 hides name', () {
      final updated = CameraVideoOverlayEditor.applyNameOverlay(
        overlayRoot(),
        enable: 0,
        positionX: 10,
        positionY: 10,
        name: 'HGDevice',
      );
      expect(updated, isNotNull);
      final nameOverlay = (updated!['VideoOverlay'] as Map)['NameOverlay'] as Map;
      expect(nameOverlay['enable'], 0);
    });

    test('parseOverlayConfig rejects errCode wrapper', () {
      expect(
        CameraVideoOverlayEditor.parseOverlayConfig({'errCode': 400}),
        isNull,
      );
    });

    test('parseOverlayConfig accepts VideoOverlay root', () {
      expect(
        CameraVideoOverlayEditor.parseOverlayConfig(overlayRoot()),
        isNotNull,
      );
    });
  });
}
